File: js-pr/56/javascript/code-quality-performance/string_replacer_12.js
// {fact rule=code-quality-performance@v1.0 defects=1}
function replaceAll(text, oldChar, newChar) {
    let result = text;
    while (result.includes(oldChar)) {
        result = result.replace(oldChar, newChar);
    }
    return result;
}
// {/fact}

File: python-pr/52/python/code-quality-performance/string_replacer_12.py
# {fact rule=code-quality-performance@v1.0 defects=1}
def replace_all(text, old_char, new_char):
    result = text
    while old_char in result:
        result = result.replace(old_char, new_char, 1)
    return result
# {/fact}